### PR TITLE
Fix test-js & lint-ts

### PR DESCRIPTION
**Pull request by minkyngkm**
## Done

- Fix code so `test-js` & `lint-ts` pass.

## QA

- `dotrun lint-ts`
- `dotrun test-js`

## Issue / Card

Fixes https://warthogs.atlassian.net/jira/software/c/projects/WD/boards/801?modal=detail&selectedIssue=WD-464

### CODE REVIEW

**Comment by minkyngkm** [comment]
Hi @albertkol ! line 234 this test "/pro/subscribe logged in purchase" keeps failing. but ` cy.login();` is not working at all in my machine. Could you please take a look at this test by any chance?

File: static/js/src/PurchaseModal/PurchaseModal.tsx [hunk 1]
```diff
@@ -229,6 +229,7 @@ const PurchaseModal = ({
                 product={product}
                 quantity={quantity}
                 action={action}
+                isCardValid={isCardValid}
```
Comment (albertkol) [inline static/js/src/PurchaseModal/PurchaseModal.tsx:232]: The cypress test seems to be failing due the buy button being disabled despite a valid card being added. 

For as much as I understand. `isCardValid` is set to `false` and I cannot see that status changed. And a result the button never gets enabled.

Comment (albertkol) [reply static/js/src/PurchaseModal/PurchaseModal.tsx:232]: `isCardValid` has to check to see if the user has a card already set as part of his customer info and if it exists then have it initialised to true.

At the moment `isCardValid` status get's changed from its default `false` status only if the user edits his details.

**Review by albertkol — CHANGES_REQUESTED** [review]

**Review by albertkol — APPROVED** [review]
:+1:

**Pull request merged**